src/util.ts

```diff
@@ -5,12 +5,23 @@ import isNode from "detect-node";
 // tslint:disable-next-line:no-var-requires
 const querystring = isNode ? require("querystring") : require("querystring-browser");
 
+interface IOriginalRequestMetadata {
+  [key: string]: any;
+  requestHeaders: any;
+  requestHost: string;
+  requestMethod: string;
+  requestPath: string;
+}
+
 export const hostIsWhitelisted =
   (whitelist: string[] | undefined, host: string | undefined, hostname: string | undefined) =>
     whitelist &&
       ((host && whitelist.indexOf(host) !== -1)
         || (hostname && whitelist.indexOf(hostname) !== -1));
 
+// TODO: buildPath and endReporter have to be made easier and simpler to use...
+//        --> perhaps part of UnmockOptions as it is in unmock-python?
```

Comment:
I'd perhaps add these comments as issues or elsewhere rather than leave them clutter the code?

Reply:
Added to JIRA